core/src/test/scala/unit/kafka/log/LogManagerTest.scala

```diff
@@ -680,4 +680,15 @@ class LogManagerTest {
     time.sleep(logConfig.fileDeleteDelayMs + 1)
     verifyMetrics(1)
   }
+
+  @Test
```

Comment:
This test may not still catch the bug introduced in #9680. 
We can continue more discussion under this review comment in #9680: https://github.com/apache/kafka/pull/9680/files#r540476157.

We should still get this PR merged though, since it fixes the problem.

Reply:
@kowshik Thanks for you explanation! It is indeed hard to give full test on the shutdown path.

> This test may not still catch the bug introduced in #9680.

you are right. I'm going to enrich the unit tests.

Reply:
For another, Your comment in ttps://github.com/apache/kafka/pull/9680/files#r540476157 is great to me. It would be better that we can make higher test coverage on that!

Reply:
Sounds good!